### ISSUE TITLE
If I delete the iCubGazeboV3 model from icub-models and I reinsert it, Gazebo hangs indefinitely

### ISSUE THREAD

**Issue by S-Dafarra** [robotology/gazebo-yarp-plugins#545]
I am using Gazebo 11.3 with the latest master of this repo. Whenever I load a model and I delete it, then the gazebo real-time factor seems stuck and I cannot add any model.

I am not sure if it is a problem of the plugins, the models, or Gazebo itself.

**Comment by traversaro** [comment]
With which model are you experiencing this?

**Comment by S-Dafarra** [comment]
> With which model are you experiencing this? On which OS?

On Ubuntu 20.04. Actually, I just run some more tests, and this seems to happen only with the ``iCubGazeboV3`` model.

**Comment by S-Dafarra** [comment]
By looking at the terminal output, I also noticed the following error:
```
Node::Advertise(): Error advertising topic [/iCub/joint_cmd]. Did you forget to start the discovery service?
```

**Comment by traversaro** [comment]
> 
> 
> By looking at the terminal output, I also noticed the following error:
> 
> ```
> Node::Advertise(): Error advertising topic [/iCub/joint_cmd]. Did you forget to start the discovery service?
> ```

Can you try to run the command `gz topic -l` to see the opened Gazebo topics for both iCub 2.5 and 3? Perhaps we can spot some differences that could point us in the right direction.

**Comment by S-Dafarra** [comment]
> > By looking at the terminal output, I also noticed the following error:
> > ```
> > Node::Advertise(): Error advertising topic [/iCub/joint_cmd]. Did you forget to start the discovery service?
> > ```
> 
> Can you try to run the command `gz topic -l` to see the opened Gazebo topics for both iCub 2.5 and 3? Perhaps we can spot some differences that could point us in the right direction.

I inserted ``iCubGazevoV2_5 (no hands)`` and obtained the following topics
```
sdafarra@iiticublap104:~$ gz topic -l
/gazebo/default/atmosphere
/gazebo/default/diagnostics
/gazebo/default/factory
/gazebo/default/factory/light
/gazebo/default/gui
/gazebo/default/gzclient_camera/cmd
/gazebo/default/iCub/chest/chest_ems_acc_eb1/imu
/gazebo/default/iCub/chest/chest_ems_acc_eb2/imu
/gazebo/default/iCub/chest/chest_ems_acc_eb3/imu
/gazebo/default/iCub/chest/chest_ems_acc_eb4/imu
/gazebo/default/iCub/chest/chest_ems_gyro_eb1/imu
/gazebo/default/iCub/chest/chest_ems_gyro_eb2/imu
/gazebo/default/iCub/chest/chest_ems_gyro_eb3/imu
/gazebo/default/iCub/chest/chest_ems_gyro_eb4/imu
/gazebo/default/iCub/chest/chest_mtb_acc_0b10/imu
/gazebo/default/iCub/chest/chest_mtb_acc_0b7/imu
/gazebo/default/iCub/chest/chest_mtb_acc_0b8/imu
/gazebo/default/iCub/chest/chest_mtb_acc_0b9/imu
/gazebo/default/iCub/chest/wrench
/gazebo/default/iCub/head/head_imu_acc_1x1/imu
/gazebo/default/iCub/head/wrench
/gazebo/default/iCub/l_ankle_1/wrench
/gazebo/default/iCub/l_ankle_2/l_foot_ft_acc_3b13/imu
/gazebo/default/iCub/l_ankle_2/l_foot_ft_gyro_3b13/imu
/gazebo/default/iCub/l_ankle_2/wrench
/gazebo/default/iCub/l_arm_ft_sensor/l_arm_ft_sensor/wrench
/gazebo/default/iCub/l_elbow_1/wrench
/gazebo/default/iCub/l_foot/l_foot_mtb_acc_10b12/imu
/gazebo/default/iCub/l_foot/l_foot_mtb_acc_10b13/imu
/gazebo/default/iCub/l_foot/wrench
/gazebo/default/iCub/l_foot_ft_sensor/l_foot_ft_sensor/wrench
/gazebo/default/iCub/l_forearm/l_forearm_mtb_acc_1b7/imu
/gazebo/default/iCub/l_forearm/l_forearm_mtb_acc_1b8/imu
/gazebo/default/iCub/l_forearm/l_forearm_mtb_acc_1b9/imu
/gazebo/default/iCub/l_forearm/wrench
/gazebo/default/iCub/l_hand/wrench
/gazebo/default/iCub/l_hip_1/wrench
/gazebo/default/iCub/l_hip_2/l_upper_leg_ft_acc_3b12/imu
/gazebo/default/iCub/l_hip_2/l_upper_leg_ft_gyro_3b12/imu
/gazebo/default/iCub/l_hip_2/wrench
/gazebo/default/iCub/l_hip_3/wrench
/gazebo/default/iCub/l_leg_ft_sensor/l_leg_ft_sensor/wrench
/gazebo/default/iCub/l_lower_leg/l_lower_leg_ems_acc_eb7/imu
/gazebo/default/iCub/l_lower_leg/l_lower_leg_ems_gyro_eb7/imu
/gazebo/default/iCub/l_lower_leg/l_lower_leg_mtb_acc_10b10/imu
/gazebo/default/iCub/l_lower_leg/l_lower_leg_mtb_acc_10b11/imu
/gazebo/default/iCub/l_lower_leg/l_lower_leg_mtb_acc_10b8/imu
/gazebo/default/iCub/l_lower_leg/l_lower_leg_mtb_acc_10b9/imu
/gazebo/default/iCub/l_lower_leg/wrench
/gazebo/default/iCub/l_shoulder_1/wrench
/gazebo/default/iCub/l_shoulder_2/wrench
/gazebo/default/iCub/l_shoulder_3/wrench
/gazebo/default/iCub/l_upper_arm/l_upper_arm_mtb_acc_1b10/imu
/gazebo/default/iCub/l_upper_arm/l_upper_arm_mtb_acc_1b11/imu
/gazebo/default/iCub/l_upper_arm/l_upper_arm_mtb_acc_1b12/imu
/gazebo/default/iCub/l_upper_arm/l_upper_arm_mtb_acc_1b13/imu
/gazebo/default/iCub/l_upper_arm/wrench
/gazebo/default/iCub/l_upper_leg/l_upper_leg_ems_acc_eb10/imu
/gazebo/default/iCub/l_upper_leg/l_upper_leg_ems_acc_eb6/imu
/gazebo/default/iCub/l_upper_leg/l_upper_leg_ems_gyro_eb10/imu
/gazebo/default/iCub/l_upper_leg/l_upper_leg_ems_gyro_eb6/imu
/gazebo/default/iCub/l_upper_leg/l_upper_leg_mtb_acc_10b1/imu
/gazebo/default/iCub/l_upper_leg/l_upper_leg_mtb_acc_10b2/imu
/gazebo/default/iCub/l_upper_leg/l_upper_leg_mtb_acc_10b3/imu
/gazebo/default/iCub/l_upper_leg/l_upper_leg_mtb_acc_10b4/imu
/gazebo/default/iCub/l_upper_leg/l_upper_leg_mtb_acc_10b5/imu
/gazebo/default/iCub/l_upper_leg/l_upper_leg_mtb_acc_10b6/imu
/gazebo/default/iCub/l_upper_leg/l_upper_leg_mtb_acc_10b7/imu
/gazebo/default/iCub/l_upper_leg/wrench
/gazebo/default/iCub/l_wrist_1/wrench
/gazebo/default/iCub/neck_1/wrench
/gazebo/default/iCub/neck_2/wrench
/gazebo/default/iCub/r_ankle_1/wrench
/gazebo/default/iCub/r_ankle_2/r_foot_ft_acc_3b14/imu
/gazebo/default/iCub/r_ankle_2/r_foot_ft_gyro_3b14/imu
/gazebo/default/iCub/r_ankle_2/wrench
/gazebo/default/iCub/r_arm_ft_sensor/r_arm_ft_sensor/wrench
/gazebo/default/iCub/r_elbow_1/wrench
/gazebo/default/iCub/r_foot/r_foot_mtb_acc_11b12/imu
/gazebo/default/iCub/r_foot/r_foot_mtb_acc_11b13/imu
/gazebo/default/iCub/r_foot/wrench
/gazebo/default/iCub/r_foot_ft_sensor/r_foot_ft_sensor/wrench
/gazebo/default/iCub/r_forearm/r_forearm_mtb_acc_2b7/imu
/gazebo/default/iCub/r_forearm/r_forearm_mtb_acc_2b8/imu
/gazebo/default/iCub/r_forearm/r_forearm_mtb_acc_2b9/imu
/gazebo/default/iCub/r_forearm/wrench
/gazebo/default/iCub/r_hand/wrench
/gazebo/default/iCub/r_hip_1/wrench
/gazebo/default/iCub/r_hip_2/r_upper_leg_ft_acc_3b11/imu
/gazebo/default/iCub/r_hip_2/r_upper_leg_ft_gyro_3b11/imu
/gazebo/default/iCub/r_hip_2/wrench
/gazebo/default/iCub/r_hip_3/wrench
/gazebo/default/iCub/r_leg_ft_sensor/r_leg_ft_sensor/wrench
/gazebo/default/iCub/r_lower_leg/r_lower_leg_ems_acc_eb9/imu
/gazebo/default/iCub/r_lower_leg/r_lower_leg_ems_gyro_eb9/imu
/gazebo/default/iCub/r_lower_leg/r_lower_leg_mtb_acc_11b10/imu
/gazebo/default/iCub/r_lower_leg/r_lower_leg_mtb_acc_11b11/imu
/gazebo/default/iCub/r_lower_leg/r_lower_leg_mtb_acc_11b8/imu
/gazebo/default/iCub/r_lower_leg/r_lower_leg_mtb_acc_11b9/imu
/gazebo/default/iCub/r_lower_leg/wrench
/gazebo/default/iCub/r_shoulder_1/wrench
/gazebo/default/iCub/r_shoulder_2/wrench
/gazebo/default/iCub/r_shoulder_3/wrench
/gazebo/default/iCub/r_upper_arm/r_upper_arm_mtb_acc_2b10/imu
/gazebo/default/iCub/r_upper_arm/r_upper_arm_mtb_acc_2b11/imu
/gazebo/default/iCub/r_upper_arm/r_upper_arm_mtb_acc_2b12/imu
/gazebo/default/iCub/r_upper_arm/r_upper_arm_mtb_acc_2b13/imu
/gazebo/default/iCub/r_upper_arm/wrench
/gazebo/default/iCub/r_upper_leg/r_upper_leg_ems_acc_eb11/imu
/gazebo/default/iCub/r_upper_leg/r_upper_leg_ems_acc_eb8/imu
/gazebo/default/iCub/r_upper_leg/r_upper_leg_ems_gyro_eb11/imu
/gazebo/default/iCub/r_upper_leg/r_upper_leg_ems_gyro_eb8/imu
/gazebo/default/iCub/r_upper_leg/r_upper_leg_mtb_acc_11b1/imu
/gazebo/default/iCub/r_upper_leg/r_upper_leg_mtb_acc_11b2/imu
/gazebo/default/iCub/r_upper_leg/r_upper_leg_mtb_acc_11b3/imu
/gazebo/default/iCub/r_upper_leg/r_upper_leg_mtb_acc_11b4/imu
/gazebo/default/iCub/r_upper_leg/r_upper_leg_mtb_acc_11b5/imu
/gazebo/default/iCub/r_upper_leg/r_upper_leg_mtb_acc_11b6/imu
/gazebo/default/iCub/r_upper_leg/r_upper_leg_mtb_acc_11b7/imu
/gazebo/default/iCub/r_upper_leg/wrench
/gazebo/default/iCub/r_wrist_1/wrench
/gazebo/default/iCub/root_link/root_link_ems_acc_eb5/imu
/gazebo/default/iCub/root_link/root_link_ems_gyro_eb5/imu
/gazebo/default/iCub/root_link/wrench
/gazebo/default/iCub/torso_1/wrench
/gazebo/default/iCub/torso_2/wrench
/gazebo/default/iCub::chest
/gazebo/default/iCub::head
/gazebo/default/iCub::l_ankle_2
/gazebo/default/iCub::l_foot
/gazebo/default/iCub::l_forearm
/gazebo/default/iCub::l_hip_2
/gazebo/default/iCub::l_lower_leg
/gazebo/default/iCub::l_upper_arm
/gazebo/default/iCub::l_upper_leg
/gazebo/default/iCub::r_ankle_2
/gazebo/default/iCub::r_foot
/gazebo/default/iCub::r_forearm
/gazebo/default/iCub::r_hip_2
/gazebo/default/iCub::r_lower_leg
/gazebo/default/iCub::r_upper_arm
/gazebo/default/iCub::r_upper_leg
/gazebo/default/iCub::root_link
/gazebo/default/joint
/gazebo/default/light/modify
/gazebo/default/log/control
/gazebo/default/log/status
/gazebo/default/model/info
/gazebo/default/model/modify
/gazebo/default/physics
/gazebo/default/physics/contacts
/gazebo/default/playback_control
/gazebo/default/pose/info
/gazebo/default/pose/local/info
/gazebo/default/pose/modify
/gazebo/default/request
/gazebo/default/response
/gazebo/default/roads
/gazebo/default/scene
/gazebo/default/selection
/gazebo/default/sensor
/gazebo/default/skeleton_pose/info
/gazebo/default/sky
/gazebo/default/undo_redo
/gazebo/default/user_camera/joy_pose
/gazebo/default/user_camera/joy_twist
/gazebo/default/user_camera/pose
/gazebo/default/user_cmd
/gazebo/default/user_cmd_stats
/gazebo/default/visual
/gazebo/default/wind
/gazebo/default/world_control
/gazebo/default/world_stats
/gazebo/performance_metrics
/gazebo/server/control
/gazebo/world/modify
```
then, I deleted it 
```
sdafarra@iiticublap104:~$ gz topic -l
/gazebo/default/atmosphere
/gazebo/default/diagnostics
/gazebo/default/factory
/gazebo/default/factory/light
/gazebo/default/gui
/gazebo/default/gzclient_camera/cmd
/gazebo/default/joint
/gazebo/default/light/modify
/gazebo/default/log/control
/gazebo/default/log/status
/gazebo/default/model/info
/gazebo/default/model/modify
/gazebo/default/physics
/gazebo/default/physics/contacts
/gazebo/default/playback_control
/gazebo/default/pose/info
/gazebo/default/pose/local/info
/gazebo/default/request
/gazebo/default/response
/gazebo/default/roads
/gazebo/default/scene
/gazebo/default/selection
/gazebo/default/sensor
/gazebo/default/skeleton_pose/info
/gazebo/default/sky
/gazebo/default/undo_redo
/gazebo/default/user_camera/joy_pose
/gazebo/default/user_camera/joy_twist
/gazebo/default/user_camera/pose
/gazebo/default/user_cmd
/gazebo/default/user_cmd_stats
/gazebo/default/visual
/gazebo/default/wind
/gazebo/default/world_control
/gazebo/default/world_stats
/gazebo/performance_metrics
/gazebo/server/control
/gazebo/world/modify
```
and added ``iCubGazeboV3 (no hands)``
```
sdafarra@iiticublap104:~$ gz topic -l
/gazebo/default/atmosphere
/gazebo/default/default/realsense_chest_depth/cmd
/gazebo/default/diagnostics
/gazebo/default/factory
/gazebo/default/factory/light
/gazebo/default/gui
/gazebo/default/gzclient_camera/cmd
/gazebo/default/iCub/chest/chest_imu_acc_1x1/imu
/gazebo/default/iCub/chest/realsense_chest_depth/image
/gazebo/default/iCub/chest/realsense_chest_rgb/cmd
/gazebo/default/iCub/chest/realsense_chest_rgb/image
/gazebo/default/iCub/chest/wrench
/gazebo/default/iCub/head/head_imu_acc_1x1/imu
/gazebo/default/iCub/head/wrench
/gazebo/default/iCub/l_ankle_1/wrench
/gazebo/default/iCub/l_ankle_2/wrench
/gazebo/default/iCub/l_arm_ft_sensor/l_arm_ft_sensor/wrench
/gazebo/default/iCub/l_elbow_1/wrench
/gazebo/default/iCub/l_foot_front/wrench
/gazebo/default/iCub/l_foot_front_ft_sensor/l_foot_front_ft_sensor/wrench
/gazebo/default/iCub/l_foot_rear/wrench
/gazebo/default/iCub/l_foot_rear_ft_sensor/l_foot_rear_ft_sensor/wrench
/gazebo/default/iCub/l_forearm/wrench
/gazebo/default/iCub/l_hand/wrench
/gazebo/default/iCub/l_hip_1/wrench
/gazebo/default/iCub/l_hip_2/wrench
/gazebo/default/iCub/l_hip_3/wrench
/gazebo/default/iCub/l_lower_leg/wrench
/gazebo/default/iCub/l_shoulder_1/wrench
/gazebo/default/iCub/l_shoulder_2/wrench
/gazebo/default/iCub/l_shoulder_3/wrench
/gazebo/default/iCub/l_upper_arm/wrench
/gazebo/default/iCub/l_upper_leg/wrench
/gazebo/default/iCub/l_wrist_1/wrench
/gazebo/default/iCub/neck_2/wrench
/gazebo/default/iCub/neck_3/wrench
/gazebo/default/iCub/r_ankle_1/wrench
/gazebo/default/iCub/r_ankle_2/wrench
/gazebo/default/iCub/r_arm_ft_sensor/r_arm_ft_sensor/wrench
/gazebo/default/iCub/r_elbow_1/wrench
/gazebo/default/iCub/r_foot_front/wrench
/gazebo/default/iCub/r_foot_front_ft_sensor/r_foot_front_ft_sensor/wrench
/gazebo/default/iCub/r_foot_rear/wrench
/gazebo/default/iCub/r_foot_rear_ft_sensor/r_foot_rear_ft_sensor/wrench
/gazebo/default/iCub/r_forearm/wrench
/gazebo/default/iCub/r_hand/wrench
/gazebo/default/iCub/r_hip_1/wrench
/gazebo/default/iCub/r_hip_2/wrench
/gazebo/default/iCub/r_hip_3/wrench
/gazebo/default/iCub/r_lower_leg/wrench
/gazebo/default/iCub/r_shoulder_1/wrench
/gazebo/default/iCub/r_shoulder_2/wrench
/gazebo/default/iCub/r_shoulder_3/wrench
/gazebo/default/iCub/r_upper_arm/wrench
/gazebo/default/iCub/r_upper_leg/wrench
/gazebo/default/iCub/r_wrist_1/wrench
/gazebo/default/iCub/root_link/wrench
/gazebo/default/iCub/torso_1/wrench
/gazebo/default/iCub/torso_2/wrench
/gazebo/default/iCub::chest
/gazebo/default/iCub::head
/gazebo/default/joint
/gazebo/default/light/modify
/gazebo/default/log/control
/gazebo/default/log/status
/gazebo/default/model/info
/gazebo/default/model/modify
/gazebo/default/physics
/gazebo/default/physics/contacts
/gazebo/default/playback_control
/gazebo/default/pose/info
/gazebo/default/pose/local/info
/gazebo/default/pose/modify
/gazebo/default/request
/gazebo/default/response
/gazebo/default/roads
/gazebo/default/scene
/gazebo/default/selection
/gazebo/default/sensor
/gazebo/default/skeleton_pose/info
/gazebo/default/sky
/gazebo/default/undo_redo
/gazebo/default/user_camera/joy_pose
/gazebo/default/user_camera/joy_twist
/gazebo/default/user_camera/pose
/gazebo/default/user_cmd
/gazebo/default/user_cmd_stats
/gazebo/default/visual
/gazebo/default/wind
/gazebo/default/world_control
/gazebo/default/world_stats
/gazebo/performance_metrics
/gazebo/server/control
/gazebo/world/modify
```
and deleted it. Then Gazebo crashed it.

This the output of ``yarp clean``
```
sdafarra@iiticublap104:~$ yarp clean
# If the cleaning process has long delays, you may wish to use a timeout, 
# specifying how long to wait (in seconds) for a test connection to a port:
#   yarp clean --timeout 5.0
Requesting list of ports from name server... 
got 39 ports
No timeout; to specify one, do "yarp clean --timeout NN.N"
Testing /clock at tcp://192.168.1.84:10002/
* No response, removing port /clock
Testing /clock/rpc at tcp://192.168.1.84:10003/
* No response, removing port /clock/rpc
Testing /icubSim/chest/inertial at tcp://192.168.1.84:10009/
* No response, removing port /icubSim/chest/inertial
Testing /icubSim/chest/inertials/measures:o at tcp://192.168.1.84:10010/
* No response, removing port /icubSim/chest/inertials/measures:o
Testing /icubSim/chest/inertials/rpc:o at tcp://192.168.1.84:10011/
* No response, removing port /icubSim/chest/inertials/rpc:o
Testing /icubSim/depthCamera/depthImage:o at tcp://192.168.1.84:10006/
* No response, removing port /icubSim/depthCamera/depthImage:o
Testing /icubSim/depthCamera/rgbImage:o at tcp://192.168.1.84:10005/
* No response, removing port /icubSim/depthCamera/rgbImage:o
Testing /icubSim/depthCamera/rpc:i at tcp://192.168.1.84:10004/
* No response, removing port /icubSim/depthCamera/rpc:i
Testing /icubSim/depthCamera_rgb/camera:o at tcp://192.168.1.84:10008/
* No response, removing port /icubSim/depthCamera_rgb/camera:o
Testing /icubSim/depthCamera_rgb/camera:o/rpc at tcp://192.168.1.84:10007/
* No response, removing port /icubSim/depthCamera_rgb/camera:o/rpc
Testing /icubSim/head/command:i at tcp://192.168.1.84:10040/
* No response, removing port /icubSim/head/command:i
Testing /icubSim/head/inertials/measures:o at tcp://192.168.1.84:10013/
* No response, removing port /icubSim/head/inertials/measures:o
Testing /icubSim/head/inertials/rpc:o at tcp://192.168.1.84:10014/
* No response, removing port /icubSim/head/inertials/rpc:o
Testing /icubSim/head/rpc:i at tcp://192.168.1.84:10039/
* No response, removing port /icubSim/head/rpc:i
Testing /icubSim/head/state:o at tcp://192.168.1.84:10041/
* No response, removing port /icubSim/head/state:o
Testing /icubSim/head/stateExt:o at tcp://192.168.1.84:10042/
* No response, removing port /icubSim/head/stateExt:o
Testing /icubSim/inertial at tcp://192.168.1.84:10012/
* No response, removing port /icubSim/inertial
Testing /icubSim/left_arm/command:i at tcp://192.168.1.84:10044/
* No response, removing port /icubSim/left_arm/command:i
Testing /icubSim/left_arm/rpc:i at tcp://192.168.1.84:10043/
* No response, removing port /icubSim/left_arm/rpc:i
Testing /icubSim/left_arm/state:o at tcp://192.168.1.84:10045/
* No response, removing port /icubSim/left_arm/state:o
Testing /icubSim/left_arm/stateExt:o at tcp://192.168.1.84:10046/
* No response, removing port /icubSim/left_arm/stateExt:o
Testing /icubSim/left_leg/command:i at tcp://192.168.1.84:10032/
* No response, removing port /icubSim/left_leg/command:i
Testing /icubSim/left_leg/rpc:i at tcp://192.168.1.84:10031/
* No response, removing port /icubSim/left_leg/rpc:i
Testing /icubSim/left_leg/state:o at tcp://192.168.1.84:10033/
* No response, removing port /icubSim/left_leg/state:o
Testing /icubSim/left_leg/stateExt:o at tcp://192.168.1.84:10034/
* No response, removing port /icubSim/left_leg/stateExt:o
Testing /icubSim/right_arm/command:i at tcp://192.168.1.84:10048/
* No response, removing port /icubSim/right_arm/command:i
Testing /icubSim/right_arm/rpc:i at tcp://192.168.1.84:10047/
* No response, removing port /icubSim/right_arm/rpc:i
Testing /icubSim/right_arm/state:o at tcp://192.168.1.84:10049/
* No response, removing port /icubSim/right_arm/state:o
Testing /icubSim/right_arm/stateExt:o at tcp://192.168.1.84:10050/
* No response, removing port /icubSim/right_arm/stateExt:o
Testing /icubSim/right_leg/command:i at tcp://192.168.1.84:10036/
* No response, removing port /icubSim/right_leg/command:i
Testing /icubSim/right_leg/rpc:i at tcp://192.168.1.84:10035/
* No response, removing port /icubSim/right_leg/rpc:i
Testing /icubSim/right_leg/state:o at tcp://192.168.1.84:10037/
* No response, removing port /icubSim/right_leg/state:o
Testing /icubSim/right_leg/stateExt:o at tcp://192.168.1.84:10038/
* No response, removing port /icubSim/right_leg/stateExt:o
Testing /icubSim/torso/command:i at tcp://192.168.1.84:10028/
* No response, removing port /icubSim/torso/command:i
Testing /icubSim/torso/rpc:i at tcp://192.168.1.84:10027/
* No response, removing port /icubSim/torso/rpc:i
Testing /icubSim/torso/state:o at tcp://192.168.1.84:10029/
* No response, removing port /icubSim/torso/state:o
Testing /icubSim/torso/stateExt:o at tcp://192.168.1.84:10030/
* No response, removing port /icubSim/torso/stateExt:o
Ignoring /slocale
Ignoring fallback
Giving name server a chance to do garbage collection.
Name server says: garbage collection done.
```

**Comment by traversaro** [comment]
> 
> 
> By looking at the terminal output, I also noticed the following error:
> 
> ```
> Node::Advertise(): Error advertising topic [/iCub/joint_cmd]. Did you forget to start the discovery service?
> ```

To clarify, this is happening when Gazebo hangs indefinitely ?

**Comment by S-Dafarra** [comment]
> > By looking at the terminal output, I also noticed the following error:
> > ```
> > Node::Advertise(): Error advertising topic [/iCub/joint_cmd]. Did you forget to start the discovery service?
> > ```
> 
> To clarify, this is happening when Gazebo hangs indefinitely ?

I payed more attention, and that happened when I clicked on it before deleting it. 

I am not sure it is related, but I just noticed that when I click on the ``iCubGazeboV3``model, a weird body appears on the left arm
![Screenshot from 2021-03-25 12-03-17](https://user-images.githubusercontent.com/18591940/112463447-8e618800-8d62-11eb-9759-c05428b33c4b.png)
This disappears as soon as I click in the gazebo world and insert it.

**Comment by traversaro** [comment]
I searched for "Did you forget to start the discovery service?" in the Gazebo codebase, and nothing showed up, so I realized that I was barking to the wrong tree. That error is printed by Ignition Transport (see https://github.com/ignitionrobotics/ign-transport/search?q=Did+you+forget+to+start+the+discovery+service%3F). So @S-Dafarra , can you to a check similar to the one done via `gz topic -l`  by using `ign topic list` ?

**Comment by S-Dafarra** [comment]
Gazebo starts:
```
sdafarra@iiticublap104:~$ ign topic --list
/iCub/joint_cmd
/introspection/eshozp/items_update
```
Insert ``iCubGazeboV2_5``
```
sdafarra@iiticublap104:~$ ign topic --list
/introspection/scgilr/items_update
sdafarra@iiticublap104:~$ 
```
Touch ``iCubGazeboV2_5``
```
sdafarra@iiticublap104:~$ ign topic --list
/iCub/joint_cmd
/introspection/scgilr/items_update
```
Delete ``iCubGazeboV2_5``
```
sdafarra@iiticublap104:~$ ign topic --list
/iCub/joint_cmd
/introspection/scgilr/items_update
```
Insert ``iCubGazeboV3``
```
sdafarra@iiticublap104:~$ ign topic --list
/default/iCub/chest/realsense_chest_depth/image
/default/iCub/chest/realsense_chest_rgb/image
/iCub/joint_cmd
/introspection/scgilr/items_update
```
Touched ``iCubGazeboV3``, appears the error above
```
sdafarra@iiticublap104:~$ ign topic --list
/default/iCub/chest/realsense_chest_depth/image
/default/iCub/chest/realsense_chest_rgb/image
/introspection/scgilr/items_update
```
Deleted ``iCubGazeboV3``
```
sdafarra@iiticublap104:~$ ign topic --list
/default/iCub/chest/realsense_chest_depth/image
/default/iCub/chest/realsense_chest_rgb/image
/introspection/scgilr/items_update
```

Gazebo is not stuck, but I cannot add any other model.

**Comment by S-Dafarra** [comment]
I just noticed by looking at the Gazebo terminal, that when deleting the model, the ``ServerInertial`` is closed twice:
```
[INFO] |yarp.devices.ServerInertial| Closing Server Inertial...
[INFO] |yarp.devices.ServerInertial| Server Intertial thread finished
[INFO] |yarp.devices.ServerInertial| Closing Server Inertial...
[INFO] |yarp.devices.ServerInertial| Server Intertial thread finished
```
while with ``iCubGazeboV2_5``, that message is printed only once.

**Comment by S-Dafarra** [comment]
>while with iCubGazeboV2_5, that message is printed only once.

This is because ``iCubGazeboV3`` has two IMUS, while ``iCubGazeboV2_5`` only one.

Actually, I just tried again, and the problem got solved. At this point, I think it either got solved on the Gazebo side (I am using version 11.5 now), or maybe some other changes in my system. I think we can close this.

**Comment by traversaro** [comment]
> At this point, I think it either got solved on the Gazebo side (I am using version 11.5 now), or maybe some other changes in my system. I think we can close this.

It could be related to https://github.com/osrf/gazebo/pull/2950 , but it is just a guess.
